src/services

```diff
@@ -724,17 +724,44 @@ p({
   "name": "ruleDAO",
   "serve": true,
   "serviceScript": """
+    dao = new foam.dao.EasyDAO.Builder(x)
+      .setEnableInterfaceDecorators(false)
+      .setOf(foam.nanos.ruler.Rule.getOwnClassInfo())
+      .setRuler(false)
+      .setInnerDAO(x.get("localRuleDAO"))
+      .build();
+
+    dao.addPropertyIndex(new foam.core.PropertyInfo[] { foam.nanos.ruler.Rule.NAME });
+    dao = new foam.dao.ValidatingDAO(x, dao);
+
+    return dao;
+  """,
+  "client": """
+    {
+      "of": "foam.nanos.ruler.Rule"
+    }
+  """
+})
+
+p({
+  "class": "foam.nanos.boot.NSpec",
+  "name": "localRuleDAO",
+  "serviceScript":
+  """
     dao = new foam.dao.EasyDAO.Builder(x)
       .setPm(true)
       .setGuid(true)
       .setOf(foam.nanos.ruler.Rule.getOwnClassInfo())
       .setJournalType(foam.dao.JournalType.SINGLE_JOURNAL)
       .setJournalName("rules")
+      .setContextualize(true)
+      .setValidated(true)
       .setRuler(false)
+      .setAuthorize(false)
```

Comment:
Why do we not authorize the dao? Imagine, if you give me access to scriptDAO I can get localRuleDAO and I can change any rules.

Reply:
If we give you access to scriptDAO you can already do whatever you want.

Reply:
The point of localDAO's is that they're unauthorized to improve performance and data accessibility by the system.